_posts/elements/2015-05-12-forms.md

```diff
@@ -0,0 +1,465 @@
+            </div>
+        </div>
+        <div class="field">
+            <label class="field-label field-required" for="email_address_default_err_helper">Email address (default, has error)</label>
+            <input class="field-input input-text has-error" type="email" name="email_address_default_err_helper" id="email_address_default_err_helper" placeholder="e.g. jane@email.com" aria-describedby="email_address_default_err_helper" required>
+            <div class="field-hint" id="email_address_default_err_helper">
+                <p>Your email address will never be shared. We promise.</p>
+            </div>
+        </div>
+        <div class="field">
+            <label class="field-label field-required" for="email_address_default_suc_helper">Email address (default, has success)</label>
+            <input class="field-input input-text has-success" type="email" name="email_address_default_suc_helper" id="email_address_default_suc_helper" placeholder="e.g. jane@email.com" aria-describedby="email_address_default_suc_helper" required>
+            <div class="field-hint" id="email_address_default_suc_helper">
+                <p>Your email address will never be shared. We promise.</p>
+            </div>
+        </div>
+        <div class="field">
+            <label class="field-label" for="full_name_simple">Your name (simple text field)</label>
+            <input class="field-input input-text input-alt" type="text" name="full_name_simple" id="full_name_simple" placeholder="e.g. Jane Doe" required>
+        </div>
+        <div class="field">
+            <label class="field-label" for="full_name_simple_err">Your name (simple text field, error)</label>
+            <input class="field-input input-text input-alt has-error" type="text" name="full_name_simple_err" id="full_name_simple_err" placeholder="e.g. Jane Doe" required>
+        </div>
+        <div class="field">
+            <label class="field-label" for="full_name_simple_suc">Your name (simple text field, success)</label>
+            <input class="field-input input-text input-alt has-success" type="text" name="full_name_simple_suc" id="full_name_simple_suc" placeholder="e.g. Jane Doe" required>
+        </div>
+        <div class="field">
+            <label class="field-label" for="full_name_simple_disabled">Your name (disabled simple text field)</label>
+            <input class="field-input input-text input-alt" type="text" name="full_name_simple_disabled" id="full_name_simple_disabled" placeholder="e.g. Jane Doe" required disabled>
+        </div>
+    </fieldset>
+</div>
+
+<h4 class="hd-6 example-set-hd">Search</h4>
+<div class="example-set">
+    <fieldset class="form-group">
+        <div class="field">
+            <label class="field-label" for="search_01">Search (with visible label)</label>
+            <input class="field-input input-text" type="search" name="search_01" id="search_01" placeholder="Search this website">
```

Comment:
This is a good demo of hidden label and icon/button combinations. Also, yay HTML5 inputs.